### PR TITLE
Document None dns policy clearly 

### LINKED ISSUE #56399: Document None dns policy clearly
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

Needs to indicate that `None` dns policy is controlled by a feature gate and by alpha. It also needs to note that only 1.9 or later Kubernetes servers will allow the None value.

Needs to be fixed for 1.9 so that public API documentation is correct.

Refer https://github.com/kubernetes/kubernetes/pull/55848#discussion_r153079403.

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Needs to indicate that None dns policy is controlled by a feature gate and by alpha. It also needs to note that only 1.9 or later Kubernetes servers will allow the None value.

Needs to be fixed for 1.9 so that public API documentation is correct.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #56399

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by feiskyer** [comment]
assign @smarterclayton

**Review by tengqm — APPROVED** [review]

File: staging/src/k8s.io/api/core/v1/types.go
Comment (MrHohn) [inline staging/src/k8s.io/api/core/v1/types.go]: A similar comment needs to be added to DNSConfig field, which is also gated by CustomPodDNS feature gate as well.

**Review by MrHohn — COMMENTED** [review]
@feiskyer Thanks for picking this up. Just one comment. (And some other autogen files need to be fixed.)

File: staging/src/k8s.io/api/core/v1/types.go
Comment (feiskyer) [reply staging/src/k8s.io/api/core/v1/types.go]: ack

**Comment by tengqm** [comment]
Please help review this as well: kubernetes/website#6441

**Comment by MrHohn** [comment]
Changes look good. From [pull-kubernetes-verify](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/56400/pull-kubernetes-verify/65293/), might need to run:
- hack/update-openapi-spec.sh
- hack/update-staging-godeps.sh
- hack/update-swagger-spec.sh

**Comment by feiskyer** [comment]
@MrHohn Thanks. Have already run those commands, but seems updating still not complete. So I'm actually run `make update` finally.

**Comment by MrHohn** [comment]
/lgtm

/assign bowei
might need milestone approval from sig lead.

**Comment by feiskyer** [comment]
ping @smarterclayton @bowei for approval.

**Comment by MrHohn** [comment]
Adding back the original approver :)
/assign @thockin 

/unassign @soltysh @caesarxuchao

**Comment by MrHohn** [comment]
Seems like it still need the `status/approved-for-milestone` label...

**Comment by feiskyer** [comment]
@thockin Could you help to approve the milestone?

**Pull request merged**
